components/AppIcon.js

```diff
@@ -35,6 +35,24 @@ const AppIcon = ({name, icon}) => {
       );
     }
 
+    // if the app is listed in popularApps, use the image specified there
+    const popularApps = Object.values(popularAppsList).filter((app) => app._id === id);
```

Comment:
Hmm, for this to work properly, the check for verifying whether an app exists on the list of the popular apps needs to be moved up. Else some apps (e.g. EarTrumpet) won't show an icon but they do have an icon on the popular list. 

Also, I think it would be much better if we can combine line 40-55 and 58-70 somehow because a lot of the code is repeated here.

Thanks!

Reply:
Let me know what you think of the second iteration. Thanks!